tests/transforms/test_decompose_transform_graph.py

```diff
@@ -370,6 +374,61 @@ def custom_decomp(theta, phi, omega, wires):
             qml.RX(-0.1, wires=[0]),
         ]
 
+    def test_decompose_with_mcm(self):
+        """Tests that circuits and decomposition rules containing MCMs are supported."""
+
+        class CustomOp(Operation):  # pylint: disable=too-few-public-methods
+
+            resource_keys = set()
+
+            @property
+            def resource_params(self) -> dict:
+                return {}
+
+        @qml.register_resources({qml.H: 1, qml.X: 1})
+        def _custom_decomp(wires, **_):
+            qml.H(wires[0])
+            m0 = qml.measure(wires[0])
```

Comment:
Is there future work planned to support returning MCM values from decompositions? I ask because of a previous discussion that we had in stand-up, where we considered that templates with MCMs in their decomposition may want to use them for other purposes. This is not a blocker for this PR.